numba/core/cpu.py

```diff
@@ -58,10 +59,16 @@ def init(self):
         # Map external C functions.
         externals.c_math_functions.install(self)
 
-        # Initialize NRT runtime
-        rtsys.initialize(self)
-
     def load_additional_registries(self):
+        # Only initialize the NRT once something is about to be compiled. The
+        # "initialized" state doesn't need to be threadsafe, there's a lock
+        # around the internal compilation and the rtsys.initialize call can be
+        # made multiple times, worse case this just gets called a bit more often
+        # than optimal.
+        if not self._nrt_initialized:
```

Comment:
`_nrt_initialized` is not needed. `rtsys.initialize` already checks for that.

Reply:
Good point, fixed in 1a07d27.